src/viur/core/bones/key.py

```diff
@@ -60,97 +59,54 @@ def singleValueFromClient(self, value, skel, bone_name, client_data):
                     )
                 ]
 
-        # Check custom validity
-        err = self.isInvalid(key)
-        if err:
-            return self.getEmptyValue(), [ReadFromClientError(ReadFromClientErrorSeverity.Invalid, err)]
-
-        if self.check:
-            if db.Get(key) is None:
-                return self.getEmptyValue(), [
-                    ReadFromClientError(
-                        ReadFromClientErrorSeverity.Invalid,
-                        "The provided key does not exist"
-                    )
-                ]
+        if not parse_only:
+            # Check custom validity
+            err = self.isInvalid(key)
+            if err:
```

Comment:
```suggestion
            
            if err := self.isInvalid(key):
```
Maybe we can simplify this